extras/rebalance/directory-rebalance.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python3
+import argparse
+import os
+import errno
+import time
+import sys
+import datetime
+import hashlib
+import logging
+
+def size_fmt(num, suffix='B'):
+    for unit in ['','Ki','Mi','Gi','Ti','Pi','Ei','Zi']:
+        if abs(num) < 1024.0:
+            return "{:3.1f} {}{}".format(num, unit, suffix)
+        num /= 1024.0
+    return "{:.2f} {}{}".format(num, 'Yi', suffix)
+
+def time_fmt(fr_sec):
+    return str(datetime.timedelta(microseconds=fr_sec * 1000000))
```

Comment:
Do we need to compute ETA time with a microsecond resolution ? It doesn't make much sense. We don't have even a second accuracy.

Reply:
timedelta takes days or seconds or microseconds to initialize, so I took microseconds to account for subsecond information as well.

Reply:
Yes, but does it make sense ? is it really meaningful to say that it's estimated to take 45.528834 seconds ? wouldn't 46 seconds be much better ? I haven't seen any tool that shows ETA values with subsecond resolution.

Reply:
So I have two tests. one for 100 MB which takes less than a second and the other is linux tree which takes minutes. So I came up with this. Do you think it is better to just print seconds for both?

Reply:
What's the value you get if you know that the migration has happened in 0.657654 seconds ? if you repeat the migration, will it take the same time ? most probably not, so it's not a very precise time. If we have a value with a precision level much much lower than a microsecond, providing a microsecond resolution seems bad to me.

If you look at how ETA is shown in tools like 'dnf' or 'yum', it always uses a second resolution. If the rebalance takes less than a second, I think reporting it as 0 seconds is ok.